server/src/test/java/io/crate/role/TransportRoleActionTest.java

```diff
@@ -353,6 +353,79 @@ public void test_alter_user_change_jwt_and_keep_password() throws Exception {
         );
     }
 
+    @Test
+    public void test_alter_user_change_password_and_keep_jwt() {
+        Map<String, Role> roleWithJwtAndPassword = new HashMap<>();
+        var oldJwtProperties = new JwtProperties("https:dummy.org", "test", null);
+        roleWithJwtAndPassword.put("John", userOf(
+            "John",
+            Set.of(),
+            new HashSet<>(),
+            getSecureHash("old-pwd"),
+            oldJwtProperties
+           )
+        );
+        var oldRolesMetadata = new RolesMetadata(roleWithJwtAndPassword);
+        Metadata.Builder mdBuilder = Metadata.builder()
+            .putCustom(RolesMetadata.TYPE, oldRolesMetadata);
+        var newPwd = getSecureHash("new-pwd");
+        boolean exists = TransportAlterRoleAction.alterRole(
+            mdBuilder,
+            "John",
+            newPwd,
+            null,
+            false,
+            false, // No reset, keep jwt
+            Map.of()
+        );
+        assertThat(exists).isTrue();
+        assertThat(roles(mdBuilder)).containsExactlyInAnyOrderEntriesOf(
+            Map.of("John", userOf(
+                "John",
+                Set.of(),
+                new HashSet<>(),
+                newPwd,
+                oldJwtProperties)
+            )
+        );
+    }
+
+    @Test
+    public void test_alter_user_reset_password_and_keep_jwt() {
```

Comment:
minor: you could combine both in one method to avoid duplicating the setup of old metadata.
Maybe with name: `test_alter_user_change_or_reset_password_and_keep_jwt

Reply:
I had to update existing util method roles() to not call build() in order not to "reset" builder. 

Otherwise I was hitting NPE due to re-use of mdBuilder.

Could you take another look?

Reply:
cool, thx!